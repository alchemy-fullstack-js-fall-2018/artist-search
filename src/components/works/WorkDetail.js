import React, { Component } from 'react';
import { getLyrics } from '../../services/fetch';

export default class WorkDetail extends Component {
  state = {
    lyrics: ''
  };

  fetchLyrics = () => {
    const { artist, title } = this.props.match.params;
    getLyrics(artist, title)
      .then(lyrics => this.setState({ lyrics }));
  };

  componentDidMount() {
    this.fetchLyrics();
  }

  render() {
    const { artist, title } = this.props.match.params;
    return (
      <div>
        <h2>{artist}</h2>
        <h3>{title}</h3>
        <pre>{this.state.lyrics}</pre>
      </div>
    );
  }
}
