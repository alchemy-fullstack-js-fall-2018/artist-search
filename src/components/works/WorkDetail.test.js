import React from 'react';
import { shallow } from 'enzyme';
import WorkDetail from './WorkDetail';

jest.mock('../../services/fetch', () => {
  return { getLyrics: () => Promise.resolve('poetic lyrics') };
});

describe ('tests for WorkDetail component', () => {
  const params = {
    artist: 'Joni Mitchell',
    title: 'Blue'
  };
  it('renders an WorkDetail component', () => {
    const wrapper = shallow(<WorkDetail match={ { params } } />);
    expect(wrapper).toMatchSnapshot();
  });
});
