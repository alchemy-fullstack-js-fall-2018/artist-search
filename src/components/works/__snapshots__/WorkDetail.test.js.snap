// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for WorkDetail component renders an WorkDetail component 1`] = `
<div>
  <h2>
    Joni Mitchell
  </h2>
  <h3>
    Blue
  </h3>
  <pre />
</div>
`;
